framework/src/node/node.ts

```diff
@@ -724,6 +726,11 @@ export class Node {
 					),
 				});
 
+				if (this._options.backup.height > 0 && this._options.backup.height === block.header.height) {
```

Comment:
Does this also ensure that the snapshot is updated when the chain reorganizes the non-finalized blocks?